src/object/obj_tx.c

```diff
@@ -2725,11 +2744,13 @@ dc_tx_per_akey_existence_task(enum obj_rpc_opc opc, daos_handle_t oh, struct dc_
 	int					 rc;
 	int					 i;
 
+	tx->tx_has_cond = 1;
```

Comment:
[Suggestion] Perhaps it'd be simpler to do this in dc_tx_attach while holding tx_lock, to avoid the question "Are there any data races?"